### ISSUE TITLE
Is it possible to use HTML in label formatter?

### ISSUE THREAD

**Issue by rdatasculptor** [JohnCoene/echarts4r#504]
Life would be a lot easier if I was allowed to use HTML in label formatting like the way that is possible in tooltips :)
Is there a way, or a hack of some kind that would make this possible?

As an example, let's compare formatting of labels and tooltips:

```
df <- data.frame(xaxis = c(1,2,3), yaxis = c(1,2,3), label= c("<span><b>hi</b></span>", "hello", "hey"))

# whithout proper dealing with HTML
df %>% 
  e_charts(xaxis) %>%
  e_scatter(yaxis, bind = label,
            label = list(show = TRUE,
                         formatter = '{b}'))

# with proper dealing with HTML
df %>% 
  e_charts(xaxis) %>%
  e_scatter(yaxis) %>%
  e_add_nested('extra', label) %>% 
  e_tooltip(triggerOn="mousemove|click",
            formatter = htmlwidgets::JS(
              'function(params){
        return params.data.extra.label;
      }'
            ))

```

As you can see, the label doesn't know how to use HTML, whilst the tooltip formatter does. 
Is there a way to make label formatter behave the same way as the tooltip formatter?

Many thanks in advance!

**Comment by beatrizmilz** [comment]
Whenever I need to configure the formatting of the text/number displayed in the tooltip, I use JS.
I would love to know when there is the possibility to use HTML as well.. :)

**Comment by rdatasculptor** [comment]
@beatrizmilz I agree, but the inconvenience I encounter is that the label can't deal with html like the way tooltip does

**Comment by JohnCoene** [comment]
I think echarts expects us to use the `rich` option for that but it's a bit convoluted.

See https://echarts.apache.org/examples/en/editor.html?c=doc-example%2Ftext-block-fragment

**Comment by rdatasculptor** [comment]
I must admit I never really tried that. My first thought was: why choose such a complex approach instead of regular html? But this example is quite clear. Thank you for pointing us to it @JohnCoene !

**Comment by JohnCoene** [comment]
Your initial intuition looks good to me: it should work. I'm not sure why echarts.js does not allow HTML in the labels.

**Comment by GitHunter0** [comment]
Hey folks, in the example below, how could I display values of `hp` and `disp` in a custom tooltip? 

Since they are in the dataframe but not in the plot, I cannot find a way to do that.

I tried the `e_add_nested` trick but didn't work. Another issue with `e_add_nested()` is that it does not have a escape hatch version.

Any light on that will be much appreciated.

```
  library(echarts4r)

  mtcars |>
    e_charts_(x = 'wt') |>
    e_line_(serie = 'qsec') |>
    e_tooltip(trigger = "axis")
```

**Comment by rdatasculptor** [comment]
How did you use e_add_nested()? The reason I ask is because it really should work

**Comment by GitHunter0** [comment]
Thank you for the follow up @rdatasculptor , I tried for example:

```r
  library(echarts4r)
  library(dplyr)

  mtcars |> 
    mutate(label = paste("Tooltip label:", hp + disp)) |> 
    e_charts_(x = 'wt') |>
    e_line_(serie = 'qsec') |>
    e_add_nested('extra', label) |> 
    e_tooltip(
      trigger = "axis",
      formatter = htmlwidgets::JS(
        'function(params){
          return params.data.extra.label;
        }'
      )
    )
```

What am I doing wrong?

**Comment by GitHunter0** [comment]
Hey @JohnCoene , could you give some light in [here](https://github.com/JohnCoene/echarts4r/issues/504#issuecomment-2407869365)? I believe it is the last piece I need to figure out to replace `plotly`.

**Comment by rdatasculptor** [comment]
@GitHunter0 , maybe like this? The trick is to use another triggerOn. Not sure why "axis" doesn't work.

```
library(echarts4r)
library(dplyr)

mtcars |> 
  mutate(label = paste("Tooltip label:", hp + disp)) |> 
  e_charts_(x = 'wt') |>
  e_line_(serie = 'qsec') |>
  e_add_nested('extra', label) |> 
  e_tooltip(
    show = TRUE,
    triggerOn="mousemove|click",
    formatter = htmlwidgets::JS(
      'function(params){
          return params.data.extra.label;
        }'
    )
  )
```

**Comment by GitHunter0** [comment]
I appreciate the hint @rdatasculptor , it's a shame `trigger = "axis"` is not working, it is so much better in most circumstances...
